tensordict/memmap.py

```diff
@@ -254,6 +254,37 @@ def from_tensor(
         out.copy_(tensor)
         return out
 
+    @classmethod
+    def empty_like(
+        cls,
+        tensor: Union[torch.Tensor, MemmapTensor, np.ndarray],
+        transfer_ownership=False,
+        prefix=None,
+        filename: Optional[str] = None,
+        mode: str = "r+",
+    ) -> MemmapTensor:
+        if isinstance(tensor, np.ndarray):
+            raise TypeError(
+                "Convert input to torch.Tensor before calling MemmapTensor."
+            )
+        device = tensor.device if hasattr(tensor, "device") else torch.device("cpu")
```

Comment:
Similarly here, given the type hints and the fact that `np.ndarray` would have already result in error, I think we can just do `tensor.dtype`.